operations/patch.go

```diff
@@ -291,21 +291,43 @@ func Patch() cli.Command {
 // checkForLargeNumFinalizedTasks retrieves an un-finalized patch document, counts the number of tasks it contains,
 // and prompts the user with a confirmation popup if the number of tasks is greater than the largeNumFinalizedTasksThreshold.
 // It returns true if the finalization process should go through, and false otherwise.
-func checkForLargeNumFinalizedTasks(ac *legacyClient, params *patchParams, patchId string) (bool, error) {
+func checkForLargeNumFinalizedTasks(rc *legacyClient, params *patchParams, patchId string) (bool, error) {
 	if params.SkipConfirm {
 		return true, nil
 	}
-	existingPatch, err := ac.GetPatch(patchId)
+	existingPatch, err := rc.GetPatch(patchId)
 	if err != nil {
 		return false, errors.Wrapf(err, "getting patch '%s'", patchId)
 	}
 	if existingPatch == nil {
 		return false, errors.Wrapf(err, "patch '%s' not found", patchId)
 	}
+	proj, err := rc.GetPatchedConfig(patchId)
+	if err != nil {
+		return false, errors.Wrapf(err, "getting patched config for patch '%s'", patchId)
+	}
+	if proj == nil {
+		return false, errors.Errorf("project config for '%s' not found", patchId)
+	}
+	generatorTasks := proj.TasksThatCallCommand(evergreen.GenerateTasksCommandName)
+	var tvPairs []model.TVPair
 	numTasksToFinalize := 0
 	for _, vt := range existingPatch.VariantsTasks {
+		for _, t := range vt.Tasks {
+			if _, ok := generatorTasks[t]; ok {
+				tvPairs = append(tvPairs, model.TVPair{
+					TaskName: t,
+					Variant:  vt.Variant,
+				})
+			}
+		}
 		numTasksToFinalize += len(vt.Tasks)
 	}
+	numEstimatedGeneratedTasks, err := rc.GetEstimatedGeneratedTasks(patchId, tvPairs)
```

Comment:
Instead of adding more REST v2 methods to the legacy REST client, could we consider using the `client.NewCommunicator` instead? That was supposed to supercede the deprecated legacy REST client, but like a lot of other code, was never really migrated fully. Ideally we'd never add more methods to the legacy client unless there's good reason to.